Allow linking directly to specific tool and persist across site

DanTup opened the pull request:
@mit-mit @filiph Please lmk what you think.

I need to test in some other browsers (looks good on Chrome + Safari on my Mac). ~~I presume opening a PR will create a stage site so I easily test from others?~~

Changes:

1. When you switch tab, it will add that tool name to the url hash fragment
2. If you link to a page with a fragment (eg `#vscode`) it'll automatically select that tab
3. Your selected tab is written to localStorage and used (if there's no url fragment)

DanTup commented:
https://dantup-flutter-staging.firebaseapp.com/

DanTup commented:
Seems to work ok for me in Chrome (Mac+Win), Safari (Mac), IE and Edge but let me know if you can break anything.

filiph commented:
That was fast! Could you please merge latest changes from master? We just launched beta and although I don't think I touched this code, I might have.

DanTup commented:
@filiph Ok, rebased and updated staging:

https://dantup-flutter-staging.firebaseapp.com/

You might need to Ctrl+Refresh - I somehow ended up with the new site but with a sidebar, it was all a bit weird (I'm not sure how things are caching - hopefully users won't see that for the live site!).

filiph commented:
LGTM. I'm asking @mit-mit for review as well.

filiph commented on js/tabs.js:
```diff
@@ -7,14 +8,36 @@ function setupToolsTabs() {
     tabContents.removeClass('current');
   }
 
+  function selectTool(id) {
```
Please add comments (what is the purpose of this function).

DanTup replied:
Done! LMK if it's not clear enough.

filiph commented:
paging @mit-mit for review, or pass

mit-mit commented:
No comments from me (I'm pretty unknowledgeable about js), but sgtm for adding support for this

DanTup commented:
@filiph LMK if you're happy for me to merge or want another set of eyes to check it.

filiph commented:
Awesome, let's land this!

DanTup commented:
Awesome, thanks! 👍 I'm not sure what the process is for deploying, but if you let me know when it's likely to be live, I'll give it a quick test.

DanTup commented:
Oh, it's done straight from master via Travis. I'll test once it's finished :)

DanTup commented:
Seems good to me (had to Ctrl+Refresh to get the new js though)
